src/Atc.Cosmos.EventStore.Cqrs/Projections/ProjectionProcessor.cs

```diff
@@ -1,29 +1,30 @@
 using System.Diagnostics.CodeAnalysis;
 using Atc.Cosmos.EventStore.Cqrs.Diagnostics;
+using Microsoft.Extensions.DependencyInjection;
 
 namespace Atc.Cosmos.EventStore.Cqrs.Projections;
 
 internal class ProjectionProcessor<TProjection> : IProjectionProcessor<TProjection>
     where TProjection : IProjection
 {
     private readonly IReadOnlyCollection<ProjectionFilter> filters;
-    private readonly IProjectionFactory projectionFactory;
     private readonly IProjectionTelemetry telemetry;
-    private readonly IProjectionMetadata projectionMetadata;
+    private readonly ProjectionMetadata<TProjection> projectionMetadata;
+    private readonly IServiceProvider serviceProvider;
     private readonly string projectionName;
 
     public ProjectionProcessor(
         IProjectionOptionsFactory optionsFactory,
-        IProjectionFactory projectionFactory,
-        IProjectionTelemetry telemetry)
+        IProjectionTelemetry telemetry,
+        ProjectionMetadata<TProjection> projectionMetadata,
```

Comment:
Note that it is the Implementation of the metadata I inject.
In old factory method it was resolved as implementation and returned as interface. 
There is no registretion of the interface in the ServiceProvider